web/packages/extension/src/player.ts

```diff
@@ -241,28 +243,31 @@ async function loadSwf(swfUrl: string) {
     });
 }
 
-window.addEventListener("pageshow", async () => {
+async function loadSwfFromHash() {
     const url = new URL(window.location.href);
     // Hash always starts with #, gotta slice that off
     const swfUrl = url.hash.length > 1 ? url.hash.slice(1) : null;
     if (swfUrl) {
+        webURL.value = swfUrl;
         await loadSwf(swfUrl);
     }
+}
+
+window.addEventListener("pageshow", async () => {
+    await loadSwfFromHash();
+});
+
+window.addEventListener("hashchange", async () => {
```

Comment:
Is it possible to pass the function in directly, or is the lambda required?

Reply:
Opened #12904.